File: BackEnd/Mmo/Mmo_Domain/ModelResponse/SupportTicketResponse.cs
namespace Mmo_Domain.ModelResponse;

public class SupportTicketResponse
{
    public int Id { get; set; }
    public int? AccountId { get; set; }
    public string Email { get; set; } = null!;
    public string? Phone { get; set; }
    public string Title { get; set; } = null!;
    public string Content { get; set; } = null!;
    public DateTime? CreatedAt { get; set; }
    public string Status { get; set; } = "OPEN";
    public AccountMiniResponse? Account { get; set; }
}

public class AccountMiniResponse
{
    public int Id { get; set; }
    public string Username { get; set; } = null!;
    public string Email { get; set; } = null!;
}

File: frontend/src/models/modelResponse/SupportTicketResponse.ts
export type TicketStatus = "PENDING" | "PROCESSING" | "CLOSED";

export interface AccountMini {
  id: number;
  username: string;
  email: string;
}

export interface SupportTicketResponse {
  id: number;
  accountId?: number;
  email: string;
  phone?: string;
  title: string;
  content: string;
  createdAt?: string;
  status?: TicketStatus;
  account?: AccountMini;
}

export interface SupportTicketListResponse {
  items: SupportTicketResponse[];
  total: number;
  page: number;
  pageSize: number;
}

export interface SupportTicketStatsResponse {
  totalTickets: number;
  pending: number;
  processing: number;
  closed: number;
}

// Response format from backend API
export interface SupportTicketApiResponse {
  value: SupportTicketResponse[];
  stats: SupportTicketStatsResponse;
}
